python/text-to-speech/test_main.py

```diff
@@ -0,0 +1,309 @@
+        """Test credentials for speech method."""
+        instance = self.get_azure_instance("WRONG_API_KEY", "WRONG_PROJECT_ID")
+        # Mock the requests.post method used in get_token.
+        with patch("requests.post") as mock_post:
+            mock_response = requests.Response
+            mock_response.text = "fake_access_token"
+            mock_post.return_value = mock_response
+            # Mock the requests.request method.
+            with patch("requests.request") as mock_request:
+                mock_request.return_value = Exception("Some error occurred")
+                # Assert the raise.
+                self.assertRaises(Exception, instance.speech, "hi", "en-US")
+
+
+class AWSTest(unittest.TestCase):
+    """AWS API Test Cases."""
+    def get_aws_instance(self, key: str, secret_key: str) -> main.AWS:
+        """Set AWS instance with request."""
+        req = MyRequest({
+            "payload": {
+                "provider": "aws",
+                "text": "hi",
+                "language": "en-US",
+            },
+            "variables": {
+                "API_KEY": key,
+                "SECRET_API_KEY": secret_key,
+            }
+        })
+        return main.AWS(req)
+
+    @parameterized.expand([
+        # Missing API_KEY.
+        (None, "123"),
+        # Missing SECRET_API_KEY.
+        ("123", None),
+        # Missing Both.
+        (None, None),
+    ])
+    def test_validate_request(self, key: str, secret_key: str) -> None:
+        """Test 'validate_request' method when fields missing or invalid."""
+        self.assertRaises(ValueError, self.get_aws_instance, key, secret_key)
+
+    def test_speech_happy(self) -> None:
+        """Test speech method for successful TextToSpeech synthesis."""
+        instance = self.get_aws_instance("123", "123")
+        # Set up mock.
+        raw_stream = io.BytesIO(DECODED_RESULT_AWS)
+        size = len(DECODED_RESULT_AWS)
+        stream_body_obj = botocore.response.StreamingBody(raw_stream, size)
+        with patch.object(boto3.Session, "client") as mock_client:
+            mock_client.return_value.synthesize_speech.return_value = {
+                "AudioStream": stream_body_obj,
+                "ContentType": "bytes",
+                "RequestCharacters": 123,
+            }
+            got = instance.speech("hi", "en-US")
+            want = DECODED_RESULT_AWS
+            # Assert the result.
+            self.assertEqual(got, want)
+
+    def test_speech_key_exception(self) -> None:
+        """Test speech method for exceptions during TextToSpeech synthesis."""
+        instance = self.get_aws_instance("123", "123")
+        # Assert the raise.
+        self.assertRaises(Exception, instance.speech, "hi", "en-US")
+
+
+class ValidateCommonTest(unittest.TestCase):
+    """Test Cases for validate_common function."""
+    def get_req(self, payload: str, variables: str, provider: str, text: str,
+                language: str) -> None:
```

Comment:
```suggestion
    def get_req(
        self,
        payload: str,
        variables: str,
        provider: str,
        text: str,
        language: str,
    ) -> None:
```

Reply:
Are you using all of these arguments?

Reply:
Yes, we did use all the args in the parameter.